### PR TITLE
refactor: remove AutoRemoveReferrers flag

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
following #680 and addressing this comment: https://github.com/oras-project/oras-go/pull/680#discussion_r1457164506

### CODE REVIEW

**Comment by shizhMSFT** [comment]
The `AutoRemoveReferrers` flag was introduced in #653

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Comment by Wwwsylvia** [comment]
Wondering if this is a `refactor`?

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: content/oci/oci.go [hunk 1]
```diff
@@ -56,19 +56,13 @@ type Store struct {
 	//   - Default value: true.
 	AutoSaveIndex bool
 
-	// AutoGC controls if the OCI store will automatically clean newly produced
-	// dangling (unreferenced) blobs during Delete() operation. For example the
-	// blobs whose manifests have been deleted. Tagged manifests will not be
+	// AutoGC controls if the OCI store will automatically clean dangling
+	// (unreferenced) blobs created by the Delete() operation. This includes the
+	// unreferenced sucessor blobs and referrers. Tagged manifests will not be
```
Comment (shizhMSFT) [inline content/oci/oci.go:61]:
```suggestion
	// unreferenced successor blobs and referrers. Tagged manifests will not be
```

File: content/oci/oci.go [hunk 2]
```diff
@@ -56,19 +56,13 @@ type Store struct {
...
 	// deleted.
```
Comment (shizhMSFT) [inline content/oci/oci.go:62]: Consider the following wording to reduce ambiguity
```suggestion
	// (unreferenced) blobs created by the Delete() operation. This includes the
	// referrers and the unreferenced successor blobs of the deleted content.
	// Tagged manifests will not be deleted.
```

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Pull request merged**